status/flowcell.py

```diff
@@ -340,57 +340,63 @@ def fetch_ont_run_stats(
-            unit = "" if "count" in metric else "bp"
-
-            if run_dict[metric] == "":
-                run_dict[metric] = 0
-
-            run_dict["_".join([metric, "str"])] = add_prefix(
-                input_int=run_dict[metric], unit=unit
+        try:
+            # Calculate new metrics
+            run_dict["basecalled_bases"] = (
+                run_dict["basecalled_pass_bases"] + run_dict["basecalled_fail_bases"]
             )
 
-            # Formatted metrics, i.e. floats transformed to predetermined unit
-            if "count" in metric:
-                unit = "M"
-                divby = 10**6
-            elif "n50" in metric:
-                unit = "Kbp"
-                divby = 10**3
-            elif "bases" in metric:
-                unit = "Gbp"
-                divby = 10**9
+            if run_dict["basecalled_pass_read_count"] <= 0:
+                run_dict["accuracy"] = 0
             else:
-                continue
-            run_dict["_".join([metric, unit])] = round(run_dict[metric] / divby, 2)
+                run_dict["accuracy"] = round(
+                    run_dict["basecalled_pass_bases"]
+                    / run_dict["basecalled_bases"]
+                    * 100,
+                    2,
+                )
+
+            """ Convert metrics from integers to readable strings and appropriately scaled floats, e.g.
+            basecalled_pass_read_count =       int(    123 123 123 )
+            basecalled_pass_read_count_str =   str(    123.12 Mbp )
+            basecalled_pass_read_count_Gbp =   float(  0.123 )
+            """
+            metrics = [
+                "read_count",
+                "basecalled_pass_read_count",
+                "basecalled_fail_read_count",
+                "basecalled_bases",
+                "basecalled_pass_bases",
+                "basecalled_fail_bases",
+                "n50",
+            ]
+            for metric in metrics:
+                # Readable metrics, i.e. strings w. appropriate units
+                unit = "" if "count" in metric else "bp"
+
+                if run_dict[metric] == "":
+                    run_dict[metric] = 0
+
+                run_dict["_".join([metric, "str"])] = add_prefix(
+                    input_int=run_dict[metric], unit=unit
+                )
+
+                # Formatted metrics, i.e. floats transformed to predetermined unit
+                if "count" in metric:
+                    unit = "M"
+                    divby = 10**6
+                elif "n50" in metric:
+                    unit = "Kbp"
+                    divby = 10**3
+                elif "bases" in metric:
+                    unit = "Gbp"
+                    divby = 10**9
+                else:
+                    continue
+                run_dict["_".join([metric, unit])] = round(run_dict[metric] / divby, 2)
+        except KeyError:
+            # Probably a run w/o basecalling
```

Comment:
Or anything really?

Reply:
If there's no other way of checking that it's without basecalling I would move this line to line 348 or similar. I think your solution is begging for difficult errors in the future when the metric format changes again.

Reply:
How about this? [e85f91e](https://github.com/SciLifeLab/genomics-status/pull/831/commits/e85f91ea9b5ba6b05470d5251edd1d26e302a185)

The command line parameters supplied at the start of the run are recorded in the database and indicate whether basecalling is enabled at sequencing start. This will however not catch runs that were basecalled retroactively, but that would require a database update performed from the sequencing directory itself which is more TACA-territory.